src/main/java/org/datadog/jmxfetch/tasks/TaskProcessor.java

```diff
@@ -37,7 +37,7 @@ public void setThreadPoolExecutor(ExecutorService executor) {
      * */
     public boolean ready() {
         ThreadPoolExecutor tpe = (ThreadPoolExecutor) threadPoolExecutor;
-        return !(tpe.getMaximumPoolSize() == tpe.getActiveCount());
+        return !tpe.isTerminated() && !(tpe.getMaximumPoolSize() == tpe.getActiveCount());
```

Comment:
just a question: is this change needed as well for the purposes of the PR, or is it an "unrelated" improvement?

Reply:
This is required because when running as daemon the underlying thread pull executor can be terminated before the shutdown hook. The problem is that starting from the shutdown hook we we check is a thread executor is ready and if not ready we recreate.

I had a false positive (ready) when actually it was terminated.

I did not had a separate method as looking at the code around I thought that adding this check tot he ready function made sense.

Totally open to that, though.

Reply:
thanks for the explanation!

Hmm, looking at where the `ready` function is used, and the case you described: does adding the condition here actually solve the issue you mentioned?

i.e.: if the underlying thread pull executor is terminated before the shutdown hook is run, the `ready` function here would return `false`, so the logic in the `App` would recreate a new thread pool executor. Is that the desired behavior?

Reply:
Thanks for looking into this. This is the expected behavior based on my investigation. The thread pool is recreated (to do some final work) and then once that work is done it will automatically exit.

I tested both in web apps (i.e. long running processes) and short lived CLI apps and this is how I could observe it behaves.

Anyway this ring a bell in you I can look better into this. Do you have a specific concern I can specifically look for?

Reply:
Hi @olivielpeau, does my answer make sense to you? Please let me know

Reply:
Ok, thanks for the explanation, makes sense!